# Data Analytics - Summative Assessment One

## Business Case
As a part of a software development team for Censored, I have been asked to conduct data-driven analysis of games on the online gaming marketplace Steam.  The purpose of this analysis is to make informed data-driven decisions on what type of game the organisation should produce in order to maximise audience reach, engagement, and cost-effectiveness.  

By examining existing games on Steam, this project aims to identify patterns in the games' popularity, reception, player engagement, tags, genres, or combinations of genres.  The aim of these insights is to create a game with the maximum reach possible to provide the greatest return on investment.  


## Data acquisition 





### Import Python Modules 🐍

In [410]:
import pandas as pd # Pandas is a module for data manipulation, it makes use of dataframe objects to store and manipulate data.
import numpy as np # Numpy is a module for numerical computing, it provides support for arrays and matrices and mathematical functions.
import matplotlib.pyplot as plt # Matplotlib is a module for data visualization.
import seaborn as sns # Seaborn is a module for statistical data visualization, built on top of Matplotlib.
import plotly.express as px # Plotly Express is a module for making interactive visualisations.

### Loading dataset 📥

In [411]:
steam_games_df = pd.read_csv("steam_games.csv")

This creates a pandas dataframe object from the origional csv file.  The reason the dataframe object are so effective for data analysis is that they have built in methods and functions which enable the user to quickly and easily inspect and transform the data.

### Dataset source and description
This analysis uses the [Steam Games Dataset](https://www.kaggle.com/datasets/fronkongames/steam-games-dataset), a public open dataset from Kaggle.  The dataset provides comprehensive information on the games sold on Steam, including commercial, technical and user engagement data.

## Initial data inspection 🧭

### Preview the data

In [412]:
steam_games_df.head(2)

,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,Discount,DLC count,About the game,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
0,2539430,Black Dragon Mage Playtest,"Aug 1, 2023",0 - 0,0,0,0.00,0,0,NaN,...,0,0,0,NaN,NaN,NaN,NaN,NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
1,496350,Supipara - Chapter 1 Spring Has Come!,"Jul 29, 2016",0 - 20000,0,0,5.24,65,0,"Springtime, April: when the cherry trees come ...",...,0,8,0,minori,MangaGamer,"Single-player,Steam Trading Cards,Steam Cloud,...",Adventure,"Adventure,Visual Novel,Anime,Cute",https://shared.akamai.steamstatic.com/store_it...,NaN


On initially looking at the dataframes head it apears that there is a missing column name.  Due to 'supported languages' being languages, 'about the game' being text and price being in the format x.xx it makes sense that the 'DiscountDLC Count' should be two columns.  This is a mistake present in the kaggle dataset.  To fix 'DiscountDLC Count' was split into 'Discount' and 'DLC Count' by a comma.  
  
Overall this isn't very usful as it only shows a few columns potentially hiding errors.

In [413]:
steam_games_df.shape

(122611, 40)

The shape of the dataframe is 40 columns and 122611 rows.  This means that there are 40 columns and 122611 rows.  40 columns are too many for this investigation, any extra colums will just confuse.  Furthermore there are way to many rows for the data to ever be manually checked by a human. 

### Check columns

In [414]:
steam_games_df.columns

Index(['AppID', 'Name', 'Release date', 'Estimated owners', 'Peak CCU',
       'Required age', 'Price', 'Discount', 'DLC count', 'About the game',
       'Supported languages', 'Full audio languages', 'Reviews',
       'Header image', 'Website', 'Support url', 'Support email', 'Windows',
       'Mac', 'Linux', 'Metacritic score', 'Metacritic url', 'User score',
       'Positive', 'Negative', 'Score rank', 'Achievements', 'Recommendations',
       'Notes', 'Average playtime forever', 'Average playtime two weeks',
       'Median playtime forever', 'Median playtime two weeks', 'Developers',
       'Publishers', 'Categories', 'Genres', 'Tags', 'Screenshots', 'Movies'],
      dtype='str')

Column names are mostly self explanatory with the exeption of a few such as 'Posative' and 'Negative' which need aditional context to make sense.

### Check datatypes

In [415]:
steam_games_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 122611 entries, 0 to 122610
Data columns (total 40 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   AppID                       122611 non-null  int64  
 1   Name                        122610 non-null  str    
 2   Release date                122611 non-null  str    
 3   Estimated owners            122611 non-null  str    
 4   Peak CCU                    122611 non-null  int64  
 5   Required age                122611 non-null  int64  
 6   Price                       122611 non-null  float64
 7   Discount                    122611 non-null  int64  
 8   DLC count                   122611 non-null  int64  
 9   About the game              114162 non-null  str    
 10  Supported languages         122611 non-null  str    
 11  Full audio languages        122611 non-null  str    
 12  Reviews                     12070 non-null   str    
 13  Header image             

Overall the dataframe has a mix of quantitative and qualitative data.  Data type is mostly correct except columns like 'release date' which should be a date.  There are also a few unusable columns due to lack of data such as 'Movies', 'Notes' or 'Score Rank'.

In [416]:
steam_games_df.describe()

,AppID,Peak CCU,Required age,Price,Discount,DLC count,Metacritic score,User score,Positive,Negative,Score rank,Achievements,Recommendations,Average playtime forever,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Movies
count,1.226110e+05,1.226110e+05,122611.000000,122611.000000,122611.000000,122611.000000,122611.000000,122611.000000,1.226110e+05,1.226110e+05,40.000000,122611.000000,1.226110e+05,1.226110e+05,122611.000000,1.226110e+05,122611.000000,0.0
mean,1.985386e+06,5.459332e+01,0.167611,4.765091,18.353663,0.545856,2.564941,0.024549,1.044986e+03,1.691974e+02,99.175000,18.087015,9.618250e+02,2.080232e+02,13.789268,1.735705e+02,14.722170,NaN
std,1.087595e+06,3.729452e+03,1.653591,12.531030,28.858970,14.516026,13.660559,1.394901,2.809173e+04,5.374645e+03,0.675107,141.493879,2.187880e+04,1.121768e+04,270.378053,1.120254e+04,294.509615,NaN
min,1.000000e+01,0.000000e+00,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000e+00,0.000000e+00,98.000000,0.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
25%,1.063175e+06,0.000000e+00,0.000000,0.550000,0.000000,0.000000,0.000000,0.000000,0.000000e+00,0.000000e+00,99.000000,0.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
50%,1.907380e+06,0.000000e+00,0.000000,2.240000,0.000000,0.000000,0.000000,0.000000,5.000000e+00,1.000000e+00,99.000000,2.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
75%,2.869560e+06,0.000000e+00,0.000000,5.240000,40.000000,0.000000,0.000000,0.000000,3.700000e+01,1.000000e+01,100.000000,19.000000,0.000000e+00,0.000000e+00,0.000000,0.000000e+00,0.000000,NaN
max,4.264350e+06,1.013936e+06,21.000000,999.980000,100.000000,3703.000000,97.000000,100.000000,7.642084e+06,1.173003e+06,100.000000,9821.000000,4.830455e+06,3.429544e+06,20088.000000,3.429544e+06,20088.000000,NaN


.describe() shows sumary statistics for numerical colums.  
Quite a few of the third quartile are 0, this is due to a large number of 0 results biasing the data some such as 'DLC Count' are probuably legitimate however for 'User score' and 'Metacritic score' no score is probuably recorded as 0.  
Columns like 'Positive', 'Negative' and 'Recomondation' columns have a median of 5, 1 and 0 respectivly but means of 1 x 10E3, 2 x 10E2 and 2 x 10E2, this large difference between the median and mean sugests a few extremly popular games with high engagment.
The 'movies' column contains no legitimate data and should be removed.

In [417]:
steam_games_df.describe(include="str")

,Name,Release date,Estimated owners,About the game,Supported languages,Full audio languages,Reviews,Header image,Website,Support url,Support email,Metacritic url,Notes,Developers,Publishers,Categories,Genres,Tags,Screenshots
count,122610,122611,122611,114162,122611,122611,12070,122530,49676,54142,100348,4256,22458,114174,113702,113658,114198,83346,116593
unique,121454,5081,14,113556,19113,3710,11884,122420,39703,35393,60515,4160,18619,70812,62684,13291,2894,77179,116483
top,Shadow of the Tomb Raider: Definitive Edition,"Oct 23, 2025",0 - 20000,Help the beautiful girls to immerse themselves...,['English'],[],"“🏆 I loved! ❤️❤️❤️” Player “🏆 Cute kittens, ea...",https://shared.akamai.steamstatic.com/store_it...,https://www.facebook.com/8FloorGames/,https://www.facebook.com/8FloorGames,info@bigfishgames.com,https://www.metacritic.com/game/pc/shadow-of-t...,This Game may contain content not appropriate ...,EroticGamesClub,BFG Entertainment,"Single-player,Family Sharing","Casual,Indie","Adventure,Casual,Hidden Object",https://shared.akamai.steamstatic.com/store_it...
freq,20,190,75404,65,55314,72730,43,20,274,328,323,20,181,231,554,23383,6700,273,20


.describe(include="str") shows sumary statistics for string colums.  
The name column contains 121454 unique vaues out of 122610 records, 'Shadow of the Tomb Raider: Definitive Edition' apears 20 times sugesting duplicate records in the dataset.  
Estimated owners shows a range of owners therfore for analysis it may be usful to split into minimum and maximum columns.    
The catagorical data columns 'Catagories', 'Genres' and 'Tags' contain comma seperated strings which requere processing before used in analysis.

### Check missing

In [418]:
missing_values = steam_games_df.isnull().sum()
missing_values

AppID                              0
Name                               1
Release date                       0
Estimated owners                   0
Peak CCU                           0
Required age                       0
Price                              0
Discount                           0
DLC count                          0
About the game                  8449
Supported languages                0
Full audio languages               0
Reviews                       110541
Header image                      81
Website                        72935
Support url                    68469
Support email                  22263
Windows                            0
Mac                                0
Linux                              0
Metacritic score                   0
Metacritic url                118355
User score                         0
Positive                           0
Negative                           0
Score rank                    122571
Achievements                       0
R

Out of the 122611 records in this data 118355 do not have a metacritic url, this shows that the vast magority of games are unknown and do not get a critic score.  

Approximatly 6 - 8 % of games are missing data in columns which would normally be expected such as 'Screenshots', 'Developers' or 'About the game'.  This may indicate there is duplicated data, demos being counted, incomplete enteries or somthing else which needs to be investigated.

## Data cleaning 🛀🏾

### Remove Duplicates

In [419]:
steam_games_df.duplicated().sum()

np.int64(0)

This code counts duplicated records.  This means there are no completly duplicated records but does not meen that there are no records for the same game.

In [420]:
steam_games_df["AppID"].duplicated().sum()

np.int64(0)

This code counts duplications in the 'AppID' column.  The games 'AppID' is unique meaning there are no repeated records based on this identifier.

In [421]:
steam_games_df["Name"].duplicated().sum()

np.int64(1156)

This code counts duplications in the 'Name' column.  There are duplicates based on 'Name' which need investigating to see if they are duplications.

In [422]:
steam_games_df[steam_games_df["Name"].duplicated(keep=False)].sort_values("Name").head(6)

,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,Discount,DLC count,About the game,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
108086,3763930,A Better Tomorrow,"Sep 30, 2025",0 - 0,0,0,0.00,0,0,"Rabies erupts without warning, rupturing the y...",...,0,0,0,WoofWoofStudio,WoofWoofStudio,"Single-player,Steam Achievements,Custom Volume...","Indie,Free To Play,Early Access",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
30085,3022780,A Better Tomorrow,"Jul 31, 2024",0 - 20000,0,0,0.00,0,0,A Better Tomorrow is a minimalist 2D city-buil...,...,0,0,0,GoragarX,GoragarX,Single-player,"Casual,Strategy,Free To Play",NaN,https://shared.akamai.steamstatic.com/store_it...,NaN
53193,557900,A Walk in the Woods,"Apr 21, 2020",0 - 20000,0,0,0.00,0,0,Most of humanity lives in urban environments. ...,...,0,0,0,Brightdawn Entertainment,Brightdawn Entertainment,"Single-player,Full controller support,Tracked ...","Indie,Simulation","Indie,Simulation,VR,Singleplayer,Family Friendly",https://shared.akamai.steamstatic.com/store_it...,NaN
62608,842420,A Walk in the Woods,"May 1, 2018",100000 - 200000,0,0,0.00,0,0,"Find Your Friends! Over the years, dozens of c...",...,0,13,0,Something Dark Studios,Something Dark Studios,"Multi-player,PvP,Online PvP,Co-op,Online Co-op...","Adventure,Free To Play,Indie","Free to Play,Horror,Multiplayer,Indie,Adventur...",https://shared.akamai.steamstatic.com/store_it...,NaN
100222,1826070,ALONE,"Dec 23, 2021",0 - 20000,0,0,8.99,0,0,It all started with a normal night shift duty....,...,0,0,0,True Nuke,True Nuke,"Single-player,Family Sharing","Action,Adventure,Indie","Psychological Horror,Exploration,Action,Hero S...",https://shared.akamai.steamstatic.com/store_it...,NaN
17238,1544540,ALONE,"May 31, 2021",50000 - 100000,0,0,0.74,0,0,ALONE is a horror adventure game. You have to ...,...,0,0,0,DakeCraft,PsychoFlux Entertainment,"Single-player,Steam Achievements,Steam Cloud,F...","Adventure,Casual,Indie","Puzzle,Female Protagonist,RPG,2D,Top-Down,Horr...",https://shared.akamai.steamstatic.com/store_it...,NaN


Looking at the 'Release Date','About the game' and 'Developer' it apears that the records are not duplicated.

In [423]:
steam_games_df[steam_games_df.duplicated(subset=["Name", "Release date", "About the game"],keep=False)].sort_values(["Name", "Release date", "About the game"]).head(6)

,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,Discount,DLC count,About the game,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
78035,34010,Alpha Protocol™,"Jun 1, 2010",500000 - 1000000,10,17,9.99,50,0,Loyalty carries a price and no one knows this ...,...,0,118,0,Obsidian Entertainment,SEGA,"Single-player,Family Sharing","Action,RPG","RPG,Stealth,Choices Matter,Action,Multiple End...",https://shared.akamai.steamstatic.com/store_it...,NaN
108996,34019,Alpha Protocol™,"Jun 1, 2010",0 - 20000,0,17,9.99,50,0,Loyalty carries a price and no one knows this ...,...,0,0,0,Obsidian Entertainment,SEGA,"Single-player,Family Sharing","Action,RPG","RPG,Stealth,Action,Choices Matter,Multiple End...",https://shared.akamai.steamstatic.com/store_it...,NaN
63287,33230,Assassin's Creed 2,"Mar 9, 2010",2000000 - 5000000,188,17,4.99,75,0,Assassin’s Creed® 2 is the follow-up to the ti...,...,473,538,411,Ubisoft Montreal,Ubisoft,Single-player,"Action,Adventure","Action,Open World,Parkour,Assassin,Stealth,Adv...",https://shared.akamai.steamstatic.com/store_it...,NaN
83437,48173,Assassin's Creed 2,"Mar 9, 2010",0 - 20000,0,17,4.99,75,0,Assassin’s Creed® 2 is the follow-up to the ti...,...,0,0,0,Ubisoft Montreal,Ubisoft,Single-player,"Action,Adventure","Action,Open World,Parkour,Adventure,Stealth,As...",https://shared.akamai.steamstatic.com/store_it...,NaN
73217,22350,BRINK,"May 9, 2011",500000 - 1000000,22,0,0.00,0,3,Brink is an immersive first-person shooter tha...,...,0,80,0,Splash Damage,Bethesda Softworks,"Single-player,Multi-player,Co-op,Steam Achieve...","Action,Free To Play","FPS,Action,Multiplayer,Parkour,Shooter,Charact...",https://shared.akamai.steamstatic.com/store_it...,NaN
73810,22360,BRINK,"May 9, 2011",0 - 20000,0,0,0.00,0,3,Brink is an immersive first-person shooter tha...,...,0,0,0,Splash Damage,Bethesda Softworks,"Single-player,Multi-player,Co-op,Steam Achieve...","Action,Free To Play","FPS,Action,Multiplayer,Parkour,Shooter,Charact...",https://shared.akamai.steamstatic.com/store_it...,NaN


This code finds the records with the same "Name", "Release date" and "About the game".  Looking at rows with the same 'Name', 'Release Date' and 'About the Game' there apears to be duplicates.  The duplicate with the lower 'AppID' apear to be real as it has a entry for 'Peak CCU' which means Peak Concurrent Players therefore as a future peak cant be lowwer than the current peak the entry with the lower 'AppID' must be more accurate.

In [424]:
steam_games_df = steam_games_df.drop_duplicates(subset=["Name", "Release date", "About the game"], keep="first")

This code removes the duplicates with the same "Name", "Release date" and "About the game" keeping the first record.

In [425]:
steam_games_df[steam_games_df.duplicated(subset=["Name", "Release date"],keep=False)].sort_values(["Name", "Release date"]).shape

(0, 40)

This code shows that there are no more duplicates with the same "Name" and "Release date".

In [426]:
steam_games_df[steam_games_df["Name"] == "Assassin's Creed 2"]

,AppID,Name,Release date,Estimated owners,Peak CCU,Required age,Price,Discount,DLC count,About the game,...,Average playtime two weeks,Median playtime forever,Median playtime two weeks,Developers,Publishers,Categories,Genres,Tags,Screenshots,Movies
63287,33230,Assassin's Creed 2,"Mar 9, 2010",2000000 - 5000000,188,17,4.99,75,0,Assassin’s Creed® 2 is the follow-up to the ti...,...,473,538,411,Ubisoft Montreal,Ubisoft,Single-player,"Action,Adventure","Action,Open World,Parkour,Assassin,Stealth,Adv...",https://shared.akamai.steamstatic.com/store_it...,NaN


This code shows that the correct duplicate was kept as can be seen by the 'Peak CCU' being 188.

### Remove Unnesesary Columns

In [427]:
steam_games_df = steam_games_df.drop(columns=['Required age', 'Price', 'Discount', 'DLC count', 'About the game',
       'Supported languages', 'Full audio languages', 'Header image', 'Website', 'Support url', 'Support email', 'Windows',
       'Mac', 'Linux', 'Score rank', 'Achievements', 'Recommendations',
       'Notes', 'Average playtime forever', 'Average playtime two weeks',
       'Median playtime two weeks', 'Developers', 'Reviews',
       'Publishers', 'Screenshots', 'Movies'])

This code removes all of the columns which are unnessisary for this investigation.

In [428]:
steam_games_df.head(2)

,AppID,Name,Release date,Estimated owners,Peak CCU,Metacritic score,Metacritic url,User score,Positive,Negative,Median playtime forever,Categories,Genres,Tags
0,2539430,Black Dragon Mage Playtest,"Aug 1, 2023",0 - 0,0,0,NaN,0,0,0,0,NaN,NaN,NaN
1,496350,Supipara - Chapter 1 Spring Has Come!,"Jul 29, 2016",0 - 20000,0,0,NaN,0,252,3,8,"Single-player,Steam Trading Cards,Steam Cloud,...",Adventure,"Adventure,Visual Novel,Anime,Cute"


### Convert Datatypes and Create Usful Columns

In [429]:
steam_games_df.info()

<class 'pandas.DataFrame'>
Index: 122500 entries, 0 to 122610
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype
---  ------                   --------------   -----
 0   AppID                    122500 non-null  int64
 1   Name                     122499 non-null  str  
 2   Release date             122500 non-null  str  
 3   Estimated owners         122500 non-null  str  
 4   Peak CCU                 122500 non-null  int64
 5   Metacritic score         122500 non-null  int64
 6   Metacritic url           4194 non-null    str  
 7   User score               122500 non-null  int64
 8   Positive                 122500 non-null  int64
 9   Negative                 122500 non-null  int64
 10  Median playtime forever  122500 non-null  int64
 11  Categories               113547 non-null  str  
 12  Genres                   114087 non-null  str  
 13  Tags                     83260 non-null   str  
dtypes: int64(7), str(7)
memory usage: 14.0 MB


'Release date' is a string and not a date.

In [430]:
steam_games_df['Release date'] = pd.to_datetime(steam_games_df['Release date'], errors='coerce')

This code changes the 'Release date' string into datetime64[us], errors='coerce' stops the code failing if it doesn' find a date and instead puts NaT which means Not a Time.

In [431]:
steam_games_df[['Estimated owners min','Estimated owners max']]= (steam_games_df['Estimated owners'].str.split(' - ', expand=True).astype(int))

This code splits the 'Estimated owners' string into 'Estimated owners min' and 'Estimated owners max' integers.

In [432]:
steam_games_df['Estimated owners'].unique()

<StringArray>
[                '0 - 0',             '0 - 20000',       '100000 - 200000',
      '500000 - 1000000',         '20000 - 50000',       '200000 - 500000',
        '50000 - 100000',     '2000000 - 5000000',   '10000000 - 20000000',
     '1000000 - 2000000',   '20000000 - 50000000',    '5000000 - 10000000',
 '100000000 - 200000000',  '50000000 - 100000000']
Length: 14, dtype: str

This code lists all of the unique strings in 'Estimated owners'.  It shows that there is a huge range within the estimation.  This is not usful so a 'Estimated owners middle' will be created.

In [433]:
steam_games_df['Estimated owners middle'] = ((steam_games_df['Estimated owners min'] + steam_games_df['Estimated owners max']) / 2).astype(int)

This code creates 'Estimated owners middle' by finding the mean of min and max.

In [434]:
steam_games_df['Interactions'] = steam_games_df['Positive'] + steam_games_df['Negative']

This code creates an 'Interactions' column by summing the 'Positive' and 'Negative' columns.

In [435]:
steam_games_df.info()

<class 'pandas.DataFrame'>
Index: 122500 entries, 0 to 122610
Data columns (total 18 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   AppID                    122500 non-null  int64         
 1   Name                     122499 non-null  str           
 2   Release date             122500 non-null  datetime64[us]
 3   Estimated owners         122500 non-null  str           
 4   Peak CCU                 122500 non-null  int64         
 5   Metacritic score         122500 non-null  int64         
 6   Metacritic url           4194 non-null    str           
 7   User score               122500 non-null  int64         
 8   Positive                 122500 non-null  int64         
 9   Negative                 122500 non-null  int64         
 10  Median playtime forever  122500 non-null  int64         
 11  Categories               113547 non-null  str           
 12  Genres                   114087 

### Split Multi Value Catagorical Data
The 'Genres' and 'Tags' columns contain catagorical data pertaining to what kind of game it is.  This needs to be spit out of the main table so it can be used in the statistics and visualisations.

In [436]:
steam_games_df['Genre / Tags'] = steam_games_df['Genres'].fillna('') + ',' + steam_games_df['Tags'].fillna('')

This code creates a 'Genre_Tags' column by combining the 'Genres' and 'Tags' columns.  This is done as the date within these columns is inconsistant or duplicated and a combined column gives the data which is needed.

In [437]:
genres_tags_df = steam_games_df[['AppID','Genre / Tags']].copy()
genres_tags_df.head(4)

,AppID,Genre / Tags
0,2539430,","
1,496350,"Adventure,Adventure,Visual Novel,Anime,Cute"
2,1034400,"Casual,Casual,Card Game,Solitaire,Puzzle,Hidde..."
3,3292190,"Casual,Indie,Simulation,"


This code copies the 'AppID' and 'Genre / Tags' column in the steam_games_df dataframe.

In [438]:
genres_tags_df['Genre / Tags'] = genres_tags_df['Genre / Tags'].str.split(',')
genres_tags_df.head(4)

,AppID,Genre / Tags
0,2539430,"[, ]"
1,496350,"[Adventure, Adventure, Visual Novel, Anime, Cute]"
2,1034400,"[Casual, Casual, Card Game, Solitaire, Puzzle,..."
3,3292190,"[Casual, Indie, Simulation, ]"


This code splits the string into a python list which will be used by the next stage.

In [439]:
genres_tags_df = genres_tags_df.explode('Genre / Tags')
genres_tags_df.head(5)

,AppID,Genre / Tags
0,2539430,
0,2539430,
1,496350,Adventure
1,496350,Adventure
1,496350,Visual Novel


This code explodes the 'Genre / Tags' column expanding the rows.

In [440]:
genres_tags_df = genres_tags_df.drop_duplicates()
genres_tags_df.head(10)

,AppID,Genre / Tags
0,2539430,
1,496350,Adventure
1,496350,Visual Novel
1,496350,Anime
1,496350,Cute
2,1034400,Casual
2,1034400,Card Game
2,1034400,Solitaire
2,1034400,Puzzle
2,1034400,Hidden Object


This code removes duplicates which is important as many of the Genres and Tags were the same.

In [441]:
genres_tags_df = genres_tags_df[genres_tags_df['Genre / Tags'] != ""]
genres_tags_df.head(5)

,AppID,Genre / Tags
1,496350,Adventure
1,496350,Visual Novel
1,496350,Anime
1,496350,Cute
2,1034400,Casual


This code removes the 'Genre / Tags' data which is an empty string, this is important as empty strings were added when the Genres and Tags columns were combined then split.

### Handle Missing Values

### Removing Intermediate Columns

In [442]:
steam_games_df = steam_games_df.drop(columns=['Estimated owners', 'Estimated owners min', 'Estimated owners max','Categories', 'Genres', 'Tags', 'Genre / Tags'])

### Verify Cleaning 🛀🏾

In [443]:
steam_games_df.sample(8)

,AppID,Name,Release date,Peak CCU,Metacritic score,Metacritic url,User score,Positive,Negative,Median playtime forever,Estimated owners middle,Interactions
122263,1702390,Voor De Kroon,2022-12-09,3,0,NaN,0,264,36,264,10000,300
66330,3912580,Lovecraft's Cat,2025-08-22,0,0,NaN,0,0,0,0,0,0
54190,1576080,SuperOffice,2024-09-02,0,0,NaN,0,18,0,0,10000,18
28336,889530,Sharp,2018-09-01,1,0,NaN,0,17,6,0,75000,23
64658,299600,Dragon Fin Soup,2015-11-03,0,0,NaN,0,168,125,80,35000,293
87502,651640,The Spook Inspectors,2019-01-25,0,0,NaN,0,13,2,0,10000,15
77087,2828320,Winter Night,2024-02-28,0,0,NaN,0,7,1,0,10000,8
50566,1989150,Thread,2022-10-18,0,0,NaN,0,19,4,0,10000,23


In [444]:
steam_games_df.info()

<class 'pandas.DataFrame'>
Index: 122500 entries, 0 to 122610
Data columns (total 12 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   AppID                    122500 non-null  int64         
 1   Name                     122499 non-null  str           
 2   Release date             122500 non-null  datetime64[us]
 3   Peak CCU                 122500 non-null  int64         
 4   Metacritic score         122500 non-null  int64         
 5   Metacritic url           4194 non-null    str           
 6   User score               122500 non-null  int64         
 7   Positive                 122500 non-null  int64         
 8   Negative                 122500 non-null  int64         
 9   Median playtime forever  122500 non-null  int64         
 10  Estimated owners middle  122500 non-null  int64         
 11  Interactions             122500 non-null  int64         
dtypes: datetime64[us](1), int64(9), 

In [445]:
genres_tags_df.info()

<class 'pandas.DataFrame'>
Index: 1299439 entries, 1 to 122610
Data columns (total 2 columns):
 #   Column        Non-Null Count    Dtype
---  ------        --------------    -----
 0   AppID         1299439 non-null  int64
 1   Genre / Tags  1299439 non-null  str  
dtypes: int64(1), str(1)
memory usage: 29.7 MB


## Exploratory data analysis

### Summary statistics

### Graphs

## Statistical analysis

## Graphical and numerical reporting

### Audience focussed charts

### Main Findings

### Recommendations